src/NexusMods.App.UI/Pages/ModLibrary/FileOrigins/FileOriginsPageViewModel.cs

```diff
@@ -13,34 +20,70 @@
 using NexusMods.App.UI.Resources;
 using NexusMods.App.UI.Windows;
 using NexusMods.App.UI.WorkspaceSystem;
+using NexusMods.CrossPlatform.Process;
 using NexusMods.Icons;
 using NexusMods.MnemonicDB.Abstractions;
 using NexusMods.Networking.Downloaders.Tasks.State;
+using NexusMods.Paths;
 using ReactiveUI;
+using ReactiveUI.Fody.Helpers;
 using RocksDbSharp;
 
 namespace NexusMods.App.UI.Pages.ModLibrary;
 
 public class FileOriginsPageViewModel : APageViewModel<IFileOriginsPageViewModel>, IFileOriginsPageViewModel
 {
     private readonly IConnection _conn;
+    private readonly IFileSystem _fileSystem;
+    private readonly ILogger<FileOriginsPageViewModel> _logger;
+    private readonly IServiceProvider _provider;
+    private readonly IFileOriginRegistry _fileOriginRegistry;
+    private readonly IOSInterop _osInterop;
     private readonly IRepository<DownloadAnalysis.Model> _dlAnalysisRepo;
     private readonly IArchiveInstaller _archiveInstaller;
 
     public ReadOnlyObservableCollection<IFileOriginEntryViewModel> FileOrigins => _fileOrigins;
-    private ReadOnlyObservableCollection<IFileOriginEntryViewModel> _fileOrigins = new([]);
+    private ReadOnlyObservableCollection<IFileOriginEntryViewModel> _fileOrigins;
+
+    // Note(sewer). Adding a [Reactive] attribute breaks the setter, don't do it please.
```

Comment:
Wdym with "breaks the setter"?

Reply:
It discards the code in the `set` section.

This normally doesn't happen with the common Fody PropertyChanged weaver, I think it's something specific to this weaver. I amended the code to note that.

Reply:
Well yeah, you wouldn't wanna have custom code inside the `set` method.